app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -84,6 +92,73 @@ public static boolean isTv(final Context context) {
         return DeviceUtils.isTV;
     }
 
+    /**
+     * Checks if the device is in desktop or DeX mode. This function should only
+     * be invoked once on view load as it is using reflection for the DeX checks.
+     * @param context the context to use for services and config.
+     * @return true if the Android device is in desktop mode or using DeX.
+     */
+    public static boolean isDesktopMode(final Context context) {
```

Comment:
this method is complex and should only be evaluated once. Is there a reason not to cache the result in a class var ? e.g. 
``` Java
private static Boolean isDesktopMode
```

Reply:
The check relies on both input devices connected and the state of desktop mode when the function is called.

Example scenario where cacheing once will result in wrong output: Changing to Desktop mode/docking a device can happen after the variable is computed and cached. When changing to desktop mode the view lifecycle hooks rerun to update the layout and is when this is computed

Reply:
oh ok. I did not think of that scenario :+1: